library/Dotfiles.re

```diff
@@ -0,0 +1,18 @@
+exception Version_Not_Provided;
+
+let versionString = fileName => {
+  Lwt_io.lines_of_file(fileName)
+  |> Lwt_stream.to_list
+  |> Lwt.map(List.hd)
+  |> Lwt.map(String.trim);
+};
+
+let getVersion = () => {
+  let%lwt cwd = Lwt_unix.getcwd();
+  try%lwt (versionString(Filename.concat(cwd, ".node-version"))) {
```

Comment:
What if `versionString` would catch the errors and provide `option(string)`, that means that `Some("file contents")` or `None` when file not found?
then, this code would be easier to follow and Reason's amazing pattern matching will help us:

```reason
let%lwt nodeVersion = Filename.concat(cwd, ".node-version") |> versionString
and nvmrc = Filename.concat(cwd, ".nvmrc") |> versionString;

switch (nodeVersion, nvmrc) {
| (None, None) => Lwt.fail(Version_Not_Provided)
| (Some(_), Some(_)) => Lwt.failwith("You have both .node-version and .nvmrc!")
| (Some(version), None)
| (None, Some(version)) => Lwt.return(version)
};
```

Reply:
or even:

```reason
let%lwt nodeVersion = Filename.concat(cwd, ".node-version") |> versionString
and nvmrc = Filename.concat(cwd, ".nvmrc") |> versionString;

switch (nodeVersion, nvmrc) {
| (None, None) => Lwt.fail(Version_Not_Provided)
| (Some(version), Some(_))
| (Some(version), None)
| (None, Some(version)) => Lwt.return(version)
};
```

Reply:
Good idea and thanks for the feedback. I think in the case where a user has both a `.node-version` and an `.nvmrc` we can be permissive if they contain matching versions (since they might just want to support multiple clients).

Whats the difference between:
```
let%lwt nodeVersion = Filename.concat(cwd, ".node-version") |> versionString
and nvmrc = Filename.concat(cwd, ".nvmrc") |> versionString;
```
and 
```
let%lwt nodeVersion = Filename.concat(cwd, ".node-version") |> versionString;
let%lwt nvmrc = Filename.concat(cwd, ".nvmrc") |> versionString;
```
The docs mention parallel binding: https://ocsigen.org/lwt/3.2.1/api/Ppx_lwt
Does this mean they will actually be run in parallel?

Reply:
Indeed, it’s like `Promise.all`!